src/patterns/SaleCard/SaleCard.tsx

```diff
@@ -42,6 +34,8 @@ export interface SaleCardProps extends ComponentProps<'div'> {
    * - 'relatedSaleTile': A variant with a smaller image size and horizontal layout on mobile devices.
    */
   variant?: SaleCardVariants;
+  /** * The <SaleCardActions /> component used to render the SaleCard CTAs. */
+  children?: React.ReactElement<typeof SaleCardActions>;
```

Comment:
curious, does this actually provide type safety for this? I didn't think that was possible.

Reply:
It sort of works, it complains if you try to put more than one child component in there, and it returns this (ugly) error: 
> This JSX tag's 'children' prop expects a single child of type 'ReactElement<ForwardRefExoticComponent<Omit<SaleCardActionsProps, "ref"> & RefAttributes<HTMLDivElement>>, string | JSXElementConstructor<...>> | undefined', but multiple children were provided.

But if you only put one child in there, it doesn't seem to care what kind of Element it is. I did some searching and yeah it looks like TypeScript doesn't yet have a reliable way to type children